pandas/core/arrays/datetimelike.py

```diff
@@ -1044,6 +1328,47 @@ def _evaluate_compare(self, other, op):
         result[mask] = filler
         return result
 
+    def _reduce(self, name, skipna=True, **kwargs):
+        op = getattr(self, name, None)
+        if op:
+            return op(skipna=skipna)
+        else:
+            return super()._reduce(name, skipna, **kwargs)
+
+    # --------------------------------------------------------------
+    # Reductions
+
+    def any(self, skipna=True):
+        values = self._values_for_reduction(skipna=skipna)
+        return len(values)
+
+    def all(self, skipna=True):
```

Comment:
is there a reason you are not dispatching to nanops? rather than rolling own here?

Reply:
I'm removing all and any.

Reply:
I've rewritten these to use nanops (will push shortly).

There's (maybe) a bug in nanmax though... 

```python
In [7]: i8 = np.array([1, pd.NaT.value, 3])

In [8]: mask = np.array([False, True, False])

In [9]: pd.core.nanops.nanmax(i8, skipna=False, mask=mask)
Out[9]: 3
```

Reply:
yeah i think originally we defined these on DTI directly because they didn't handle tz's. So this path may not have been exercised.